### ISSUE TITLE
[Wayland] issue with mouse cursor 'point of action' 

### ISSUE THREAD

**Issue by mroe1** [fltk/fltk#521]
**Describe the bug**
When changing the cursor type the 'point of action'  keeps the same with Wayland. With 1.4.0/x11 it works fine.
The  'click point' should be different , e.g  upper left for 'Arrow' and in the middle for 'Cross'. 
This has several critical effects for using in an UI. Hitting a point with a cross arrow is nearly impossible.

**To Reproduce**
Have added a modified cursor demo with a tiny click-spot for toggle between FL_CURSOR_ARROW and FL_CURSOR_CROSS.
Click the tiny spot with 'Arrow' mouse cursor, The type change to 'Cross' and you can see the difference.
You will see the bad effect esp. if you move the mouse a bit and try to hit the button with the 'Cross'-cursor again.

[cursor_pointer_test.cxx.txt](https://github.com/fltk/fltk/files/9845152/cursor_pointer_test.cxx.txt)

**Expected behavior**
'point of action' changes per cursor type as e.g x11 does.

**FLTK Version**
 - Version: [e.g. 1.4.0]
 - Installed as package from: [e.g. Debian 11.4 Bullseye]
 - Downloaded and self-built from: Git  master (1.4)

**Operating System / Platform:**
 - OS Version: Ubuntu 20.04,22.04, OpenSuse Leap 15.4

**Linux/Unix Runtime, if applicable:**
 - Wayland

**Additional context**

**Comment by ManoloFLTK** [comment]
@mroe1  Thanks for this report. I've looked into that in detail. In fact, the cursor's 'point of action' is correctly accounted for in general (but not always, as you report, see below). That it's accounted for can be seen, for example, choosing FL_CURSOR_CROSS in the "Cursor" Fl_Choice and clicking in or near the "Cursor" wide rectangle to open the menu window: the cross center rather than any other cursor point determines whether the menu window opens.

But, as you report, this doesn't hold right after clicking with the arrow cursor on your "toggle click" small button. It's enough, though, to move the FL_CURSOR_CROSS cursor out of the window or in the window titlebar and back in after, for the cross cursor to be correctly set as a pointing device: once this is done, move the cross center above the small button and click, the button's callback correctly responds.

Unfortunately, I don't find how to improve that.

**Comment by mroe1** [comment]
IMO a user cannot instinctively workaround this problem [by e.g. leaving the window and come back ] because he is not even know what happening. Simply his mouse or the app does not what he expects.

I have a feeling it has something to do with the reported x/y mouse position in events. Means that there is a correction done somewhere that is not happen in all cases or is deferred.

-------------------------
The following may not directly caused by the original issue but is clearly related to cursor type change too. 
May be it helps when thinking about : 
Have also seen that if we change the cursor type on the fly within a specific area and dragging the mouse hin and fort, the visualization of a tooltip  gets offset-ted to his 'normal' location.
There is no good test case to reproduce yet . Will try to put something together that makes this clear.

**Comment by mroe1** [comment]
OK, have to revert what i told about event x/y shift. 
But i have a better crude test case attached. 
Here you can see the error occur when you click&drag the mouse out of the box and then release it .
It is going well when simply only move the mouse through the box and cursor change therefore.

[cursor_change_test.cxx.txt](https://github.com/fltk/fltk/files/9854610/cursor_change_test.cxx.txt)

**Comment by ManoloFLTK** [comment]
@mroe1  I think I got it! Your 2 test programs should work OK with commit e0c3fe9. Please, close issue if you confirm.

**Comment by mroe1** [comment]
> @mroe1 I think I got it! Your 2 test programs should work OK with commit [e0c3fe9](https://github.com/fltk/fltk/commit/e0c3fe939fcc5f42370a79372d833c25415f349e). Please, close issue if you confirm.

yes that fixed that, 
but unfortunately it is a regression of "[Wayland] Drag and Drop is broken for Wayland with custom handler #516" 
This is not working anymore because of that. Reproducible with DnDinsideWinV2

**Comment by ManoloFLTK** [comment]
I've seen too that the fix here affected DnD. I believe it's now OK with commit f9acdb6. It's necessary to memorize 2 different serial numbers!

**Comment by mroe1** [comment]
Thanks @ManoloFLTK  is all fine now .
